spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -643,6 +679,142 @@ def main(argv: Sequence[str]):
+            status, output = run_proc(cmd_split, verbose, is_sct_binary=True)
+            printv(output, verbose)
+            # Compute slicewise mean in Z to ensure symmetry
+            img = Image('step10Warp_Z.nii.gz')
+            out = img.copy()
+            out.data = slicewise_mean(out.data, 2)
+            out.save('step10Warp_Z_mean.nii.gz')
+            # Merge warp back together
+            cmd_split = ['sct_image', '-i',
+                         'step10Warp_X.nii.gz', 'step10Warp_Y.nii.gz', 'step10Warp_Z_mean.nii.gz',
+                         '-omc', '-o', 'step10Warp_zmean.nii.gz']
+            status, output = run_proc(cmd_split, verbose, is_sct_binary=True)
+            printv(output, verbose)
+
+            # Apply transformation
+            sct_apply_transfo.main(argv=[
+                '-i', ftmp_data,
+                '-o', add_suffix(ftmp_data, '_Rootlets'),
+                '-d', ftmp_template,
+                '-w', 'step10Warp_zmean.nii.gz',
+                '-x', 'linear',
+                '-v', '0',
+            ])
+            ftmp_data = add_suffix(ftmp_data, '_Rootlets')
+            sct_apply_transfo.main(argv=[
+                '-i', ftmp_seg,
+                '-o', add_suffix(ftmp_seg, '_Rootlets'),
+                '-d', ftmp_template,
+                '-w', 'step10Warp_zmean.nii.gz',
+                '-x', 'linear',
+                '-v', '0',
+            ])
+            ftmp_seg = add_suffix(ftmp_seg, '_Rootlets')
+
+            sct_apply_transfo.main(argv=[
+                '-i', ftmp_rootlet,
+                '-o', add_suffix(ftmp_rootlet, '_Rootlets'),
+                '-d', ftmp_template,
+                '-w', 'step10Warp_zmean.nii.gz',
+                '-x', 'nn',
+                '-v', '0',
+            ])
+            ftmp_rootlet = add_suffix(ftmp_rootlet, '_Rootlets')
+
+            printv('\nConcatenate transformations: curve --> straight --> affine --> rootlets', verbose)
+            dimensionality = len(Image("template.nii").hdr.get_data_shape())
+            cmd = [
+                'isct_ComposeMultiTransform',
+                str(dimensionality),
+                'warp_curve2straightAffine.nii.gz',
+                '-R', 'template.nii',
+                'step10Warp_zmean.nii.gz',
+                'warp_curve2straightAffine.nii.gz',
+            ]
+            status, output = run_proc(cmd, verbose=verbose, is_sct_binary=True)
+            if status != 0:
+                raise RuntimeError(f"Subprocess call {cmd} returned non-zero: {output}")
+
+            # INVERSE WARPING FIELD
+            # Average perslice warping field
+            cmd_split = ['sct_image', '-i', 'step10InverseWarp.nii.gz', '-mcs']
+            status, output = run_proc(cmd_split, verbose, is_sct_binary=True)
+            printv(output, verbose)
+            # Compute slicewise mean in Z to ensure symmetry
+            img = Image('step10InverseWarp_Z.nii.gz')
```

Comment:
idem; see https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4647/files#r1890792804

```suggestion
            # Compute slicewise mean in Z to ensure symmetry
            printv('\nComputing slicewise mean in Z to ensure symmetry....', verbose)
            img = Image('step10InverseWarp_Z.nii.gz')
```